src/main/java/software/amazon/smithy/lsp/ext/Completions.java

```diff
@@ -104,4 +157,121 @@ private static CompletionItem createCompletion(String s, CompletionItemKind kind
         return ci;
     }
 
+    private static SmithyCompletionItem smithyCompletionItem(CompletionItem item, String namespace, String name) {
+        return new SmithyCompletionItem(item, namespace, name);
+    }
+
+    private static List<CompletionItem> createTraitCompletions(Shape shape, Model model, CompletionItemKind kind) {
+        List<CompletionItem> completions = new ArrayList<>();
+        completions.add(createTraitCompletion(shape, model, kind));
+        // Add a default completion for structure shapes with members.
+        if (shape.isStructureShape() && !shape.members().isEmpty()) {
+            if (shape.members().stream().anyMatch(member -> member.hasTrait(RequiredTrait.class))) {
+                // If the structure has required members, add a default with empty parens.
+                completions.add(createCompletion(shape.getId().getName() + "()", kind));
+            } else {
+                // Otherwise, add a completion without any parens.
+                completions.add(createCompletion(shape.getId().getName(), kind));
+            }
+
+        }
+        return completions;
+    }
+
+    private static CompletionItem createTraitCompletion(Shape shape, Model model, CompletionItemKind kind) {
+        String traitBody = shape.accept(new TraitBodyVisitor(model));
+        // Strip outside pair of brackets from any structure traits.
+        if (traitBody.charAt(0) == '{') {
+            traitBody = traitBody.substring(1, traitBody.length() - 1);
+        }
+        if (shape.members().isEmpty()) {
+            return createCompletion(shape.getId().getName(), kind);
+        }
+        return createCompletion(shape.getId().getName() + "(" + traitBody + ")", kind);
+    }
+
+    private static final class TraitBodyVisitor extends ShapeVisitor.Default<String> {
+        private final Model model;
+
+        TraitBodyVisitor(Model model) {
+            this.model = model;
+        }
+
+        @Override
+        protected String getDefault(Shape shape) {
+            return "";
+        }
+
+        @Override
+        public String bigDecimalShape(BigDecimalShape shape) {
+            return "\"\"";
```

Comment:
Maybe this is a completion thing, that might warrant a comment, but why an empty string for these numbers/timestamp that'd be a number value and not a string value in the IDL?

Reply:
For this visitor, I set any shapes that could be represented as strings, per this trait node values [table](https://awslabs.github.io/smithy/1.0/spec/core/model.html#trait-node-values), as `""`.

Is that table out of date?

Since those are likely to be set as a number value, maybe they should use the default case of an empty string like `byte`, `short`, `integer` and `long`.

Reply:
At least `float` and `double` should default, since the only valid string values are the edge cases. I think this should use number values for the arbitrary precision as well, since our tooling handles them properly.

I could see leaving `timestamp` as a string value, since it'll be more readable in the model than a Unix epoch.

Reply:
Updated `float`, `double`, `bigDecimal` and `bigInteger` to default in 25f2d8e2b1d21b658d6a9132f0511fbf61fb1aa0.